packages/super-components/components/msc-billing-tile/src/msc-billing-tile/msc-billing.types.ts

```diff
@@ -0,0 +1,343 @@
+      interval: number;
+      duration?: string;
+      minimumQuantity?: number;
+      maximumQuantity?: number;
+      minimumRepeat?: number;
+      maximumRepeat?: number;
+      price?: {
+        currencyCode?: CurrencyCode;
+        text?: string;
+        value?: number;
+      };
+      priceInUcents?: number;
+      pricingMode: string;
+      pricingType: 'consumption' | 'purchase' | 'rental';
+      engagementConfiguration?: {
+        defaultEndAction: Strategy;
+        duration?: string;
+        type?: 'periodic' | 'upfront';
+      };
+    };
+    group: { id: number } | null;
+    lifecycle: {
+      current: {
+        pendingActions: LifecycleAction[];
+        // Datetime format
+        terminationDate?: string | null;
+        // Datetime format
+        creationDate?: string;
+        state:
+          | 'active'
+          | 'error'
+          | 'rupture'
+          | 'terminated'
+          | 'toRenew'
+          | 'unpaid'
+          | 'unrenewed';
+      };
+      capacities: {
+        actions: LifecycleAction[];
+      };
+    };
+    renew: {
+      current: {
+        mode: 'automatic' | 'manual';
+        // Datetime format
+        nextDate: string;
+        period: Period;
+      };
+      capacities: {
+        mode: ('automatic' | 'manual')[];
+      };
+    };
+    engagement?: {
+      endDate?: string;
+      endRule?: { possibleStrategies: Strategy[]; strategy?: Strategy };
+    };
+    engagementRequest?: { requestDate?: string; pricingMode?: string };
+  };
+  customer: {
+    contacts: {
+      customerCode: string;
+      type: 'administrator' | 'billing' | 'technical';
+    }[];
+  };
+  route?: {
+    path: string;
+    url: string;
+    vars: {
+      key: string;
+      value: string;
+    }[];
+  };
+  ressource?: {
+    displayName: string;
+    name: string;
+    state:
+      | 'active'
+      | 'deleted'
+      | 'suspended'
+      | 'toActivate'
+      | 'toDelete'
+      | 'toSuspend';
+    product: {
+      name: string;
+      description: string;
+    };
+    resellingProvider?: 'ovh.ca' | 'ovh.eu' | null;
+  };
+  parentServiceId?: number;
+  serviceId?: number;
+  tags?: string[];
+};
+
+export type DomainProperties = {
+  parentService: null | { name: string; type: '/allDom' }[];
+  dnssecSupported: boolean;
+  hostSupported: boolean;
+  // Datetime
+  lastUpdate: string;
+  whoisOwner: string;
+  suspensionState?: 'not_suspended' | 'suspended';
+  glueRecordIpv6Supported: boolean;
+  // Service name
+  domain: string;
+  glueRecordMultiIpSupported: boolean;
+  transferLockStatus:
+    | 'locked'
+    | 'locking'
+    | 'unavailable'
+    | 'unlocked'
+    | 'unlocking';
+  owoSupported: boolean;
+  nameServerType: 'hosted' | 'external';
+  offer: 'diamond' | 'gold' | 'platinum';
+  state:
+    | 'deleted'
+    | 'dispute'
+    | 'expired'
+    | 'ok'
+    | 'outgoing_transfer'
+    | 'pending_delete'
+    | 'pending_installation'
+    | 'restorable';
+};
```

Comment:
**nitpick:** This is very specific to Domain product and should not be maintained in super-component as the component can be used in multiple products.

Reply:
Why domain product specific logic can't be included in super-components ?
They are included in the specification